app/src/main/java/com/weatherxm/ui/devicesettings/helium/DeviceSettingsHeliumActivity.kt

```diff
@@ -275,4 +271,44 @@ class DeviceSettingsHeliumActivity : BaseActivity() {
 
         model.getDeviceInformation(this)
     }
+
+    private fun handleSplitRewards(data: RewardSplitsData?) {
+        if (data?.hasSplitRewards() == true) {
+            binding.rewardSplitCard.visible(true)
+            binding.rewardSplitDesc.text = getString(R.string.reward_split_desc, data.splits.size)
+            val rewardSplitAdapter = RewardSplitStakeholderAdapter(data.wallet, true)
+            binding.recyclerRewardSplit.adapter = rewardSplitAdapter
+            rewardSplitAdapter.submitList(data.splits)
+
+            val stakeHolderValue = if (model.isStakeholder(data)) {
+                AnalyticsService.ParamValue.STAKEHOLDER_LOWERCASE.paramValue
+            } else {
+                AnalyticsService.ParamValue.NON_STAKEHOLDER.paramValue
+            }
+            analytics.trackEventViewContent(
+                AnalyticsService.ParamValue.REWARD_SPLITTING_DEVICE_SETTINGS.paramValue,
+                contentId = null,
+                Pair(
+                    AnalyticsService.CustomParam.DEVICE_STATE.paramName,
+                    AnalyticsService.ParamValue.REWARD_SPLITTING.paramValue
+                ),
+                Pair(AnalyticsService.CustomParam.USER_STATE.paramName, stakeHolderValue)
+            )
+        } else {
+            val stakeHolderValue = if (model.device.isOwned()) {
+                AnalyticsService.ParamValue.STAKEHOLDER_LOWERCASE.paramValue
+            } else {
+                AnalyticsService.ParamValue.NON_STAKEHOLDER.paramValue
+            }
+            analytics.trackEventViewContent(
+                AnalyticsService.ParamValue.REWARD_SPLITTING_DEVICE_SETTINGS.paramValue,
+                contentId = null,
+                Pair(
+                    AnalyticsService.CustomParam.DEVICE_STATE.paramName,
+                    AnalyticsService.ParamValue.NO_REWARD_SPLITTING.paramValue
+                ),
+                Pair(AnalyticsService.CustomParam.USER_STATE.paramName, stakeHolderValue)
+            )
```

Comment:
A function with argument `REWARD_SPLITTING` or `NO_REWARD_SPLITTING` would be more clear